tasks/resolve_reference.rb

```diff
@@ -65,9 +65,8 @@ def load_remote_statefile(opts)
   end
 
   def load_local_statefile(opts)
-    dir = opts[:dir]
     filename = opts.fetch(:statefile, 'terraform.tfstate')
-    File.read(File.expand_path(File.join(dir, filename)))
+    File.read(File.expand_path(File.join(opts[:dir], filename), opts[:_boltdir]))
```

Comment:
I think that should be relative to the `dir` provided.

Reply:
Maybe I don't understand TF remote state, but I assumed it lived on a separate machine where the `boltdir` isn't really meaningful? Is that not the case?

Reply:
The state itself is stored on a remote backend, but you still have a `terraform.tfstate` file that defines where/what the backend is. TF needs to read this file to load the remote state.

Reply:
oh, wait the remote state file. @beechtom is correct. You still need to execute from the `dir` provided (the info for finding the backend is in that context). 
my bad.

Reply:
Ah, ok, I see - it changes to that directory to execute the `terraform state pull`. I think if we're expanding for local storage it makes sense to do that for remote as well. I'm less certain we should be expanding either of these relative to the Boltdir in the first place, since I doubt most folks organize their boltdir near their tf infra files. But I also assume most people are using absolute paths for this, and consistency with the rest of Bolt seems more predictable longterm for users.